be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/DataBuilder.scala

```diff
@@ -72,4 +85,20 @@ object DataBuilder {
     case ActionType.ADD_BROADCAST => AddBroadcastChirp.buildFromJson(payload)
     case _ => throw new ProtocolException(s"Unknown action '$action' encountered while creating a Social Media Data")
   }
+  /**
+    * Builds a message payload after passing a schema validation check
+    *
+    * @param payload payload to build
+    * @param validator one of the validators at [[DataSchemaValidator]] to valid the schema of the payload
+    * @param buildFromJson the data builder
+    * @param errMsg error message to include in description in case of error
+    * @return built MessageData or throws an exceptition in case of schema failure
+    */
+  @throws(classOf[ProtocolException])
+  private def buildOrReject(payload: String)(validator: String => Try[Unit])(buildFromJson: String => MessageData )(errMsg: String): MessageData = {
+      validator(payload) match {
+        case Success(_) => buildFromJson(payload)
+        case Failure(e) => throw new ProtocolException(s"$errMsg: ${e.getMessage}")
```

Comment:
Exception message is not very descriptive with regard to the source of the issue (i.e. buildFromJson)

Reply:
The description is in errMsg.
The buildLaoData methode makes a call like this 
`buildOrReject(payload)(DataSchemaValidator.validateSchema(ObjectType.LAO)(ActionType.CREATE))(CreateLao.buildFromJson)("CreateLao data could not be parsed")`

Reply:
Understood, my bad :)